File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        int number = 0;
        
        Map<Character, Integer> numerals = new HashMap<>();
        numerals.put('I', 1);
        numerals.put('V', 5);
        numerals.put('X', 10);
        numerals.put('L', 50);
        numerals.put('C', 100);
        numerals.put('D', 500);
        numerals.put('M', 1000);

        for (int i = 0; i < s.length(); i++) {
            if(i>0 && numerals.get(s.charAt(i-1)) < numerals.get(s.charAt(i))) {
                number = number - numerals.get(s.charAt(i-1))*2 + numerals.get(s.charAt(i));
            } else {
                number += numerals.get(s.charAt(i));
            }
        }
        return number;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.go
func romanToInt(s string) int {
    number := 0

    numerals := map[rune]int {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000,
    }

    for i := 0; i < len(s); i++ {
        if i > 0 && numerals[[]rune(s)[i-1]] < numerals[[]rune(s)[i]] {
            number = number - numerals[[]rune(s)[i-1]]*2 + numerals[[]rune(s)[i]]
        } else {
            number += numerals[[]rune(s)[i]]
        }
    }

    return number
}